### ISSUE TITLE
Property method "fget" is incompatible

### ISSUE THREAD

**Issue by Hocnonsense** [microsoft/pyright#9059]
**Describe the bug**
When I use `pyright` to check each of my file, nothing happened; 
But when I check the folder, error occurs.

**Code or Screenshots**
runs and outputs from [github action](https://github.com/Hocnonsense/pymummer/actions/runs/10986457454/job/30499844953):
```log
/home/runner/work/pymummer/pymummer/pymummer/pair.py
  /home/runner/work/pymummer/pymummer/pymummer/pair.py:45:9 - error: "seq2" incorrectly overrides property of same name in class "AlignContig2"
    Property method "fget" is incompatible
      Parameter 1 name mismatch: base parameter is named "self", override parameter is named "cls" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations 
```

But the line pyright point out is (45-47):
```python
    def __call__(self, cls):
        self.cls = cls
        return self
```
all pythin files are in https://github.com/Hocnonsense/pymummer/tree/dd7b5c1a4099aa6ad7846876c08f1e21c6b5a0bf

**VS Code extension or command-line**
Using the command-line tool, version 1.1.381

**Comment by Hocnonsense** [comment]
Sure, i've used the default settings. 
I'd just create a new environment with mamba: `mamba create -n pyright "python=3.12" pyright`, and clone the reposity again `git clone https://github.com/Hocnonsense/pymummer; cd pymummer`. Then I've got the `pair.py:45:9 - error` again.

Do you have any idea? Thanks for your help!

**Comment by erictraut** [comment]
Thanks, I was able to repro the problem and fix it. Pyright is correct in generating this error, but it's reporting it in the wrong location. It should be associated with the `DeltaContig2.seq2` method.

This will be fixed in the next release.

**Comment by Hocnonsense** [comment]
Thanks!

By the way, I'm puzzled about why there is an issue with `DeltaContig2.seq2` method.
I think it is similar to 
```python
class A:
    @property
    def a(self) -> int:
        return 1


class A1(A):
    @property
    def a(self):
        return 2
```
, but both `.seq2` property returns a `Pair[SeqRecord]` object.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.382.
